src/angular-selector.js

```diff
@@ -0,0 +1,70 @@
+import { Selector } from 'testcafe';
+
+export default Selector(complexSelector => {
+    function validateSelector (selector) {
+        if (selector !== void 0 && typeof selector !== 'string')
+            throw new Error(`If the selector parameter is passed it should be a string, but it was ${typeof selector}`);
+    }
+
+    function getComponentTag (debugElement) {
+        return debugElement.nativeElement.tagName.toLowerCase();
+    }
+
+    function getTagList (componentSelector) {
+        return componentSelector
+            .split(' ')
+            .filter(el => !!el)
+            .map(el => el.trim().toLowerCase());
+    }
+
+    function filterNodes (rootDebugElement, tags) {
+        const foundNodes = [];
+
+        function walkDebugElements (debugElement, tagIndex, checkFn) {
+            if (checkFn(debugElement, tagIndex)) {
+                if (tagIndex === tags.length - 1) {
+                    foundNodes.push(debugElement.nativeElement);
+                    return;
+                }
+
+                tagIndex++;
+            }
+
+            for (const childDebugElement of debugElement.children)
+                walkDebugElements(childDebugElement, tagIndex, checkFn);
+        }
+
+        walkDebugElements(rootDebugElement, 0, (debugElement, tagIndex) => {
+            if (!debugElement.componentInstance)
+                return false;
+
+            return tags[tagIndex] === getComponentTag(debugElement);
+        });
+
+        return foundNodes;
+    }
+
+    validateSelector(complexSelector);
+
+    // NOTE: We support only applications with a single root element
```

Comment:
Why? We can look into every root component or make root component optional.

Reply:
I've investigated this area and didnt't find angular application with multiple roots.
So, now we don't support applications with multiple roots

Reply:
The phrase `We don't support an application with multiple roots` means that you can't use this plugin for such applications. In fact, you can. We have a note about this in the readme: `If your application has multiple roots, AngularSelector will return the first root returned by the window.getAllAngularRootElements function`.
So it's better to write comment like `If there are multiple roots on the page we find a target in the first root only`.

Reply:
Ok